nft_ingester/src/program_transformers/token/mod.rs

```diff
@@ -77,21 +77,22 @@ pub async fn handle_token_program_account<'a, 'b, 'c>(
                 .await?;
 
             if let Some((asset, data)) = asset_with_data {
+                let mut update = true;
+                if let Some(data) = data {
+                    if data.metadata_url.is_empty() {
+                        update = false;
+                        statsd_count!("token_account.empty_metadata", 1);
+                    } else {
+                        statsd_count!("token_account.non_empty_metadata", 1);
+                    }
+                }
                 // will only update owner if token account balance is non-zero
-                if ta.amount > 0 {
+                if ta.amount > 0 && update {
```

Comment:
I don't follow. Doesn't this break indexing?

Why would we not index token account changes for an NFT without a metadata url?